### DISCUSSION TITLE
[Bug] Scoop launches Windows "How do you want to open this file?" prompt (scoop.ps1 missing?)

### DISCUSSION THREAD

**Discussion by ehawman** [ScoopInstaller/Scoop#5628] [Help and Questions]
#### Current Behavior
Running scoop {anything} results in Windows launching the "How do you want to open this file?" prompt. 
![image](https://github.com/ScoopInstaller/Scoop/assets/52979227/38d8ce34-0b99-49ac-9e1a-7c14357ce9c1)

![image](https://github.com/ScoopInstaller/Scoop/assets/52979227/fac15aac-68a6-4795-820c-b2b9708c2c29)

#### Expected Behavior
Scoop does scoopy CLI things.

#### Additional context/output
I believe this happened during an update process (`scoop update --all`). scoop.ps1 is not in `%USERPROFILE%\scoop\shims\`.

Perhaps related, perhaps not, but my shims for git and nushell are missing `[error 2147942402 (0x80070002) when launching '%USERPROFILE%\scoop\shims\nu.exe']`, `git: The term 'git' is not recognized as a name of a cmdlet`...

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:** 5.1.22000.2003 or 7.4.0.preview.5

**Additional software:** nu, git

#### Scoop Configuration

```json
{
    "last_update":  "2023-09-01T09:39:15.0393902-05:00",
    "scoop_branch":  "master",
    "scoop_repo":  "https://github.com/ScoopInstaller/Scoop"
}
```

**Comment by HUMORCE** [comment]
Related on StackOverflow:

https://stackoverflow.com/questions/71684323/powershell-asks-how-do-you-want-to-open-this-file-when-executing-command
https://stackoverflow.com/questions/45418557/running-npm-on-powershell-asks-how-do-you-want-to-open-this-file-command-lin

**Reply by ehawman** [reply]
Reading those threads, they had inadvertently created an empty file in System32 (or another place on PATH). Windows attempted to run this extensionless file, and popped up the HDYWTOTF prompt.

In my case, I have the following shims:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ehawman/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ehawman/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



LastWriteTime | Length | Name
-- | -- | --
11/14/2022 18:50 | 126464 | 7z.exe
8/29/2023 12:01 | 58 | 7z.shim
11/14/2022 18:50 | 126464 | 7zfm.exe
8/29/2023 12:01 | 60 | 7zfm.shim
11/14/2022 18:50 | 126464 | 7zg.exe
8/29/2023 12:01 | 59 | 7zg.shim
9/1/2023 9:42 | 63 | bash.shim
11/14/2022 18:50 | 126464 | bat.exe
8/29/2023 12:06 | 58 | bat.shim
11/14/2022 18:50 | 126464 | btm.exe
8/29/2023 12:07 | 61 | btm.shim
11/14/2022 18:50 | 126464 | bw.exe
8/29/2023 12:07 | 67 | bw.shim
11/14/2022 18:50 | 126464 | caffeine.exe
8/29/2023 12:07 | 70 | caffeine.shim
11/14/2022 18:50 | 126464 | caffeine64.exe
8/29/2023 12:07 | 70 | caffeine64.shim
8/29/2023 12:40 | 157 | clink
8/29/2023 12:40 | 118 | clink.cmd
11/14/2022 18:50 | 126464 | croc.exe
8/29/2023 12:09 | 60 | croc.shim
11/14/2022 18:50 | 126464 | curlie.exe
8/29/2023 12:09 | 64 | curlie.shim
11/14/2022 18:50 | 126464 | db browser for sqlcipher.exe
8/29/2023 12:16 | 89 | db browser for sqlcipher.shim
11/14/2022 18:50 | 126464 | db browser for sqlite.exe
8/29/2023 12:16 | 86 | db browser for sqlite.shim
11/14/2022 18:50 | 126464 | delta.exe
8/29/2023 12:09 | 62 | delta.shim
11/14/2022 18:50 | 126464 | duf.exe
8/29/2023 12:10 | 58 | duf.shim
11/14/2022 18:50 | 126464 | dust.exe
8/29/2023 12:10 | 60 | dust.shim
11/14/2022 18:50 | 126464 | exiftool(-k).exe
8/29/2023 12:10 | 72 | exiftool(-k).shim
11/14/2022 18:50 | 126464 | exiftool.exe
8/29/2023 12:10 | 68 | exiftool.shim
11/14/2022 18:50 | 126464 | fastfetch.exe
8/29/2023 12:10 | 70 | fastfetch.shim
11/14/2022 18:50 | 126464 | fd.exe
8/29/2023 12:10 | 56 | fd.shim
11/14/2022 18:50 | 126464 | ffmpeg.exe
8/29/2023 12:10 | 75 | ffmpeg.shim
11/14/2022 18:50 | 126464 | ffplay.exe
8/29/2023 12:10 | 75 | ffplay.shim
11/14/2022 18:50 | 126464 | ffprobe.exe
8/29/2023 12:10 | 76 | ffprobe.shim
11/14/2022 18:50 | 126464 | flashfetch.exe
8/29/2023 12:10 | 71 | flashfetch.shim
11/14/2022 18:50 | 126464 | fzf.exe
8/29/2023 12:10 | 58 | fzf.shim
11/14/2022 18:50 | 126464 | genpass.exe
8/29/2023 12:10 | 64 | genpass.shim
9/1/2023 9:42 | 62 | git.shim
9/1/2023 9:42 | 63 | git-bash.shim
9/1/2023 9:42 | 66 | git-gui.shim
9/1/2023 9:42 | 63 | gitk.shim
11/14/2022 18:50 | 126464 | innounp.exe
8/29/2023 12:10 | 66 | innounp.shim
11/14/2022 18:50 | 126464 | kageant.exe
8/29/2023 12:10 | 64 | kageant.shim
11/14/2022 18:50 | 126464 | kitty.exe
8/29/2023 12:10 | 62 | kitty.shim
11/14/2022 18:50 | 126464 | kittygen.exe
8/29/2023 12:10 | 65 | kittygen.shim
11/14/2022 18:50 | 126464 | klink.exe
8/29/2023 12:10 | 62 | klink.shim
11/14/2022 18:50 | 126464 | kscp.exe
8/29/2023 12:10 | 61 | kscp.shim
11/14/2022 18:50 | 126464 | ksftp.exe
8/29/2023 12:10 | 62 | ksftp.shim
11/14/2022 18:50 | 126464 | less.exe
8/29/2023 12:10 | 60 | less.shim
11/14/2022 18:50 | 126464 | lesskey.exe
8/29/2023 12:10 | 63 | lesskey.shim
11/14/2022 18:50 | 126464 | lsd.exe
8/29/2023 12:10 | 58 | lsd.shim
11/14/2022 18:50 | 126464 | micro.exe
8/29/2023 12:10 | 62 | micro.shim
8/29/2023 12:10 | 127 | neofetch
8/29/2023 12:10 | 177 | neofetch.cmd
8/29/2023 12:48 | 56 | nu.shim
11/14/2022 18:50 | 126464 | procs.exe
8/29/2023 12:14 | 62 | procs.shim
11/14/2022 18:50 | 126464 | rclone.exe
8/29/2023 12:15 | 64 | rclone.shim
11/14/2022 18:50 | 126464 | rg.exe
8/29/2023 12:15 | 61 | rg.shim
11/14/2022 18:50 | 126464 | rufus.exe
8/29/2023 12:15 | 62 | rufus.shim
9/1/2023 9:42 | 65 | scalar.shim
9/1/2023 9:39 | 349 | scoop
11/14/2022 18:50 | 126464 | sd.exe
8/29/2023 12:15 | 56 | sd.shim
9/1/2023 9:42 | 61 | sh.shim
8/29/2023 12:16 | 331 | sudo
8/29/2023 12:16 | 330 | sudo.cmd
8/29/2023 12:16 | 225 | sudo.ps1
9/1/2023 9:42 | 66 | tig.shim
11/14/2022 18:50 | 126464 | win32diskimager.exe
8/29/2023 12:16 | 84 | win32diskimager.shim
11/14/2022 18:50 | 126464 | zoom.exe
8/29/2023 12:17 | 60 | zoom.shim
11/14/2022 18:50 | 126464 | zoxide.exe
8/29/2023 12:17 | 64 | zoxide.shim



</body>

</html>

Note that our problem children are the ones that don't have an executable with them (git, nu, scoop, sh).

I think it has to do with the mysterious disappearance of those executable files. I think I'll try installing Scoop to a custom location then dropping those files on top of this installation.

**Reply by ehawman** [reply]
This did the trick. I copied only the missing executables over from a working installation. Git didn't work, but I was able to use Scoop to un/re-install it and now everything appears to be behaving.

Are there update logs I can look for to find out if scoop is accidentally zapping those executable files during update or if it's something else? I've already combed through my antivirus and it claims innocence.

**Reply by rashil2000** [reply]
> Are there update logs I can look for

Scoop does not store any logs.

Skimming through the directory listing above, it looks like a lot of executable binaries and executable scripts have been zapped. Judging by previous reports of similar kind, this is most likely the work of an over-enthusiastic antivirus scan run.
